### PR TITLE
fix(core): add complex test for routing list

### PR DESCRIPTION
**Pull request by jacobowitz**
This adds a test to verify that https://github.com/jina-ai/jina/issues/2500 got fixed as a side effect of dynamic routing (https://github.com/jina-ai/jina/pull/2551).
Not sure we need this test actually as part of the test suite, but I was writing it anyway to verify the behaviour and I think there are no other tests doing this.

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1200479679796904/1200517178620445) by [Unito](https://www.unito.io)

### CODE REVIEW

**Comment by hanxiao** [comment]
actually i belive #2500 is fixed due to https://github.com/jina-ai/jina/pull/2750/files#diff-d9002011181954eeec0b9b94a9d08f7bc8028338275a35bea7823aa07d08041dL33

File: tests/unit/flow/test_flow.py [hunk 1]
```diff
@@ -747,3 +749,71 @@ def test_flow_common_kwargs():
 def test_flow_set_asyncio_switch_post(is_async):
     f = Flow(asyncio=is_async)
     assert inspect.isasyncgenfunction(f.post) == is_async
+
+
```
Comment (bwanglzu) [inline tests/unit/flow/test_flow.py:753]: can you mark it as a slow test?
```suggestion
pytest.mark.slow
```

**Review by bwanglzu — COMMENTED** [review]
can you benchmark these two test if slow mark it as `pytest.mark.slow`?

**Comment by jacobowitz** [comment]
> can you benchmark these two test if slow mark it as `pytest.mark.slow`?

I checked its 3 seconds. Lets not mark it as slow as its rather nice to execute locally (no docker etc involved)

**Review by bwanglzu — APPROVED** [review]
LGTM!

**Pull request merged**